RHDEVDOCS 5985 changes for HashiCorp Vault integration (#79150)
openshift/openshift-docs: modules/op-creating-mounting-kms-authentication-token-secret.adoc

// This module is included in the following assemblies:
// * secure/using-tekton-chains-for-openshift-pipelines-supply-chain-security.adoc

:_mod-docs-content-type: PROCEDURE
[id="creating-mounting-kms-authentication-token-secret_{context}"]
= Creating and mounting the KMS authentication token secret

> You can provide the authentication token for the KMS server (the value of `VAULT_TOKEN`) using a secret. You must create this secret, mount it on the {tekton-chains} controller, and set the `signers.kms.auth.token-dir` parameter to the directory where the secret is mounted.

Reviewer: kindly consider updating to "You can provide the authentication token for the KMS provider (for example the value of VAULT_TOKEN if the KMS provider is Hashicorp Vault) using a secret."

> You can provide the authentication token for the KMS server (for example, the the value of `VAULT_TOKEN` if the KMS provider is Hashicorp Vault) using a secret. You must create this secret, mount it on the {tekton-chains} controller, and set the `signers.kms.auth.token-dir` parameter to the directory where the secret is mounted.

Reviewer 2: This has been changed now, user can create a secret with any key, and `VAULT_TOKEN`  is not mandatory now https://github.com/tektoncd/chains/blob/main/docs/config.md#kms-oidc-and-spire-configuration. Also the field is changed from `signers.kms.auth.token-dir` to `signers.kms.auth.token-path`

And this is the related PR https://github.com/tektoncd/chains/pull/1167

Author: @PuneetPunamiya please clarify. Does this mean that the `signers.kms.auth.token-path` parameter must specify the full path and name of the file to read? Or does it still specify a directory but any file from that directory is read?

If one follows the examples currently in the doc, the token is in the file named `/etc/kms-secrets/KMS_AUTH_TOKEN` . In this example, should `signers.kms.auth.token-path` be set to `/etc/kms-secrets` or `/etc/kms-secrets/KMS_AUTH_TOKEN` ?

Reviewer 2: Yeah the parameter means that now it specifies the full path and name of the file to read.

In the example you mentioned above if `signers.kms.auth.token-path` is specified then it's value should be `/etc/kms-secrets/KMS_AUTH_TOKEN`

> You can provide the authentication token for the KMS server (for example, the the value of `VAULT_TOKEN` if the KMS provider is Hashicorp Vault) using a secret. You must create this secret, mount it on the {tekton-chains} controller, and set the `signers.kms.auth.token-path` parameter to the full pathname of the authentication token file.

Reviewer 3: ```suggestion
You can provide the authentication token for the KMS server using a secret. For example, the the value of `VAULT_TOKEN` if the KMS provider is Hashicorp Vault. You must create this secret, mount it on the {tekton-chains} controller, and set the `signers.kms.auth.token-path` parameter to the full pathname of the authentication token file.
```

.Prerequisites

> * You installed the `oc` command-line utility.

Reviewer 3: ```suggestion
* You installed the {oc-first} utility.
```

> * You are logged in to your {OCP} claster with administrative rights for the `tekton-chains` namespace.

Reviewer: typo - replace claster to cluster

.Procedure

> . Create a secret named `kms-secrets` with the `KMS_AUTH_TOKEN` file that contains the authentication token for the KMS server by entering the following command:

Reviewer 2: Same as above
User can create secret with any key now, we can put this as an example

+
[source, terminal]
----

> $ oc create secret generic kms-secret -n tekton-chains \ #

Reviewer: secret name "kms-secret" mismatches with secret name in config section. Please set it to "kms-secrets" so that it is consistent.

Reviewer 2: The secret name can be anything not just `kms-secret` but user need to make sure that same secretName is added in the chains controller deployment

Reviewer: This comment on naming was in the context of example spec from line number 28-55. Just to ensure consistency for the reader to relate to the example. Otherwise name can be anything

Author: I made the name consistent as kms-secrets - I want to keep the procedure as simple as possible so would want to avoid explaining the part about other names

  --from-file=KMS_AUTH_TOKEN=<path>/KMS_AUTH_TOKEN # <1>
----
<1> The full path and name of the `KMS_AUTH_TOKEN` file that contains the authentication token for the KMS server. 

> . In the `TektonConfig` custom resource (CR), in the `chain` section, configure mounting the secret on the {tekton-chains} controller and set the `signers.kms.auth.token-dir` parameter to the directory where the secret is mounted, as shown in the following example.

Reviewer 2: ```suggestion
. In the `TektonConfig` custom resource (CR), in the `chain` section, configure mounting the secret on the {tekton-chains} controller and set the `signers.kms.auth.token-path` parameter to the path where the secret is mounted, as shown in the following example.
```

> . In the `TektonConfig` custom resource (CR), in the `chain` section, configure mounting the secret on the {tekton-chains} controller and set the `signers.kms.auth.token-path` parameter to the full pathname of the authentication token file, as shown in the following example.

Reviewer 3: ```suggestion
. In the `TektonConfig` custom resource (CR), in the `chain` section, configure mounting the secret on the {tekton-chains} controller and set the `signers.kms.auth.token-path` parameter to the full pathname of the authentication token file, as shown in the following example:
```

+
.Example configuration for mounting the `kms-secrets` secret
[source,yaml]
----
apiVersion: operator.tekton.dev/v1
kind: TektonConfig
metadata:
  name: config
spec:
# ...
  chain:
    disabled: false

>     signers.kms.auth.token-dir: /etc/kms-secrets

Reviewer 2: ```suggestion
    signers.kms.auth.token-path: /etc/kms-secrets
```
